### PR TITLE
add embedded py interpreter for flow, based on VisIt's implementation

### PR DESCRIPTION
**Pull request by cyrush**

### CODE REVIEW

**Comment by cyrush** [comment]
@mclarsen  i think this is ready, do you want to give it a quick review?

File: src/ascent/runtimes/ascent_main_runtime.cpp [hunk 1]
```diff
@@ -410,27 +410,69 @@ AscentRuntime::ConvertExtractToFlow(const conduit::Node &extract,
...
+       {
+         ASCENT_ERROR("broadcast of python script source failed");
+       }
+       // replace file param with source that includes actual script
+       params.remove("file");
+       params["source"] = n_py_src;
+     }
+#endif
+  
+    // todo, inspect args, if passed via file, read on root proc and broadcast
```
Comment (mclarsen) [inline src/ascent/runtimes/ascent_main_runtime.cpp]: Isn't this what the above code does?

File: src/ascent/runtimes/ascent_main_runtime.cpp [hunk 2]
```diff
@@ -410,27 +410,69 @@ AscentRuntime::ConvertExtractToFlow(const conduit::Node &extract,
...
     ASCENT_ERROR("Unrecognized extract type "<<extract["type"].as_string());
   }
  
   if(w.graph().has_filter(extract_name))
   {
-    ASCENT_INFO("Duplicate extract name "<<extract_name
-                <<" original is being overwritted");
+    ASCENT_INFO("Duplicate extract filter named " 
+                << "\"" << extract_name << "\"" 
+                << " original is being overwritten");
```
Comment (mclarsen) [inline src/ascent/runtimes/ascent_main_runtime.cpp]: We need to get rid of these checks or make them ASCENT_ERRORS. Flow will not let us add duplicates anyway and will fail. Here we can give a more understandable error message than flow.

Comment (cyrush) [reply src/ascent/runtimes/ascent_main_runtime.cpp]: thanks, i'll tackle these.

Not sure how to guard against the specific error about using the wrong python, b/c if thats the case we begin on a bad footing.

**Pull request merged**